### PR TITLE
Fixing #1254 - Remove support for X-Reported-With header

### PR DESCRIPTION
**Pull request by zoepage**
r? @miketaylr

### CODE REVIEW

**Comment by miketaylr** [comment]
In terms of fixing the issue, you nailed it. I have a small Python doubt (which means a bug I think I introduced... will comment, because you can probably fix it in this PR).

File: webcompat/views.py [hunk 1]
```diff
@@ -179,14 +179,7 @@ def create_issue():
...
     form['ua_header'] = request.headers.get('User-Agent')
-    # Currently we support either a src GET param, or X-Reported-With header
-    # to track where the report originated from.
-    # See https://github.com/webcompat/webcompat.com/issues/1254 to track
-    # supporting only the src param
-    if session.get('src'):
-        form['reported_with'] = session.pop('src')
-    else:
-        form['reported_with'] = request.headers.get('X-Reported-With', 'web')
+    form['reported_with'] = session.pop('src')
```
Comment (miketaylr) [inline webcompat/views.py]: So `src` will end up in the `session` if there was a `src` param:

https://github.com/webcompat/webcompat.com/blob/a5b5d84a06cbf1564247208ac8bf8e825134326d/webcompat/views.py#L166

(for all add-on and "report site issue "reported issues, that gets added for the user).

If this was a web-reported issue, `src` won't be in the session and a `KeyError` will be raised (which means the page will explode, as is).

[`dict#pop`](https://docs.python.org/2/library/stdtypes.html#dict.pop) does allow you to specify a default value if the key isn't there though, as the second arg. 

So I think we want `session.pop('src', 'web')` -- which should mean if there wasn't a `src` param added by an add-on, we should set `reported_with` to `web`.

Does that make sense, @zoepage? Anything I'm overlooking?

Comment (zoepage) [inline webcompat/views.py]: @miketaylr 

so basically `form['reported_with'] = session.pop('src')` in line 182 would get the additional param, right? (Just to be sure and don't waste your time :) )

**Comment by miketaylr** [comment]
r+, thanks!

File: webcompat/views.py (hunk 1 above)
Comment (karlcow) [inline webcompat/views.py]: @zoepage For a descriptive example of what @miketaylr is talking about
https://github.com/webcompat/webcompat.com/pull/1255#discussion_r93997278

**Pull request merged**